src/main/java/it/gov/pagopa/pu/debtpositions/service/installmentsync/operation/InstallmentSynchronizeCreateBalanceService.java

```diff
@@ -0,0 +1,115 @@
+package it.gov.pagopa.pu.debtpositions.service.installmentsync.operation;
+
+import it.gov.pagopa.pu.classification.dto.generated.AssessmentsRegistry;
+import it.gov.pagopa.pu.classification.dto.generated.AssessmentsRegistryStatus;
+import it.gov.pagopa.pu.classification.dto.generated.PagedModelAssessmentsRegistry;
+import it.gov.pagopa.pu.debtpositions.connector.classification.service.AssessmentsRegistryService;
+import it.gov.pagopa.pu.debtpositions.connector.classification.service.BalanceService;
+import it.gov.pagopa.pu.debtpositions.dto.generated.InstallmentSynchronizeDTO;
+import it.gov.pagopa.pu.debtpositions.exception.custom.InvalidValueException;
+import it.gov.pagopa.pu.debtpositions.model.DebtPositionTypeOrg;
+import it.gov.pagopa.pu.debtpositions.repository.DebtPositionTypeOrgRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.BooleanUtils;
+import org.springframework.stereotype.Service;
+import org.apache.commons.lang3.StringUtils;
+
+import java.time.LocalDate;
+import java.util.Set;
+
+@Service
+@Slf4j
+public class InstallmentSynchronizeCreateBalanceService {
+  private final DebtPositionTypeOrgRepository debtPositionTypeOrgRepository;
+  private final AssessmentsRegistryService assessmentsRegistryService;
+  private final BalanceService balanceService;
+
+  private static final String OPERATING_YEAR = String.valueOf(LocalDate.now().getYear());
+  private static final AssessmentsRegistryStatus ASSESSMENTS_REGISTRY_STATUS = AssessmentsRegistryStatus.ACTIVE;
+
+  public InstallmentSynchronizeCreateBalanceService(DebtPositionTypeOrgRepository debtPositionTypeOrgRepository, AssessmentsRegistryService assessmentsRegistryService, BalanceService balanceService) {
+    this.debtPositionTypeOrgRepository = debtPositionTypeOrgRepository;
+    this.assessmentsRegistryService = assessmentsRegistryService;
+    this.balanceService = balanceService;
+  }
+
+  public String createBalance(InstallmentSynchronizeDTO installmentSynchronizeDTO, String accessToken) {
+    if (StringUtils.isNotBlank(installmentSynchronizeDTO.getBalance())) {
+      return installmentSynchronizeDTO.getBalance();
+    }
```

Comment:
from a service which is call `createBalance` I don't expect a logic where if it exists it does nothing...
I would instead change the logic:
* on the mapper I would verify if the balance on the Installment exists:
* if yes It would just use it
* otherwise it would call a Service called getDebtPositionTypeDefaultBalance